src/main/java/practice/StreamPractice.java

```diff
@@ -23,7 +29,11 @@ public int findMinEvenNumber(List<String> numbers) {
      * But before that subtract 1 from each element on an odd position (having the odd index).
      */
     public Double getOddNumsAverage(List<Integer> numbers) {
-        return 0D;
+        return IntStream.range(0, numbers.size())
+                .map(i -> i % 2 == 1 ? numbers.get(i) - 1 : numbers.get(i))
+                .filter(i -> i % 2 == 1)
+                .average()
+                .orElseThrow(() -> new NoSuchElementException("There are no elements"));
```

Comment:
will throw that exception for you
```suggestion
                .getAsDouble();
```